consensus/src/msg_verifier.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    block_storage::tracing::{observe_block, BlockStage},
+    monitor,
+    payload_manager::PayloadManager,
+    round_manager::{UnverifiedEvent, VerifiedEvent},
+};
+use anyhow::{bail, Context};
+use aptos_channels::aptos_channel;
+use aptos_consensus_types::common::Author;
+use aptos_logger::{error, warn, SecurityEvent};
+use aptos_types::epoch_state::EpochState;
+use move_core_types::account_address::AccountAddress;
+use std::{
+    hash::Hash,
+    mem::{discriminant, Discriminant},
+    sync::Arc,
+};
+use tokio::runtime::Handle;
+
+// Verifies the consensus messages and forwards them to the corresponding components - done by having a
+// dedicated thread pool for signature verification so that we don't block the consensus runtime.
+pub struct ConsensusMsgVerifier {
```

Comment:
I don't see why we need this struct, also not sure if a tokio runtime is better than a rayon pool since all verification is synchronous work.

what I imagine was just to use the ThreadManager.get_exe_cpu_pool().spawn(verify_work)